sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/parser/AstBuilder.scala

```diff
@@ -659,12 +670,21 @@ class AstBuilder extends SqlBaseBaseVisitor[AnyRef] with SQLConfHelper with Logg
         AttributeReference("value", StringType)()), true)
     }
 
+    val plan = visitCommonSelectQueryClausePlan(relation,
+      lateralView,
+      transformClause.namedExpressionSeq,
+      whereClause,
+      aggregationClause,
+      havingClause,
+      windowClause,
+      isDistinct = false)
+
     // Create the transform.
     ScriptTransformation(
-      expressions,
+      Seq(UnresolvedStar(None)),
```

Comment:
I thinks its better to leave comments about how the analyzer handle this `UnresolvedStar` later.

Reply:
Done